lib/packwerk/cli.rb

```diff
@@ -10,15 +10,23 @@ class Cli
         configuration: T.nilable(Configuration),
         out: T.any(StringIO, IO),
         err_out: T.any(StringIO, IO),
-        style: Packwerk::OutputStyle
+        style: Packwerk::OutputStyle,
+        offenses_formatter: T.nilable(Packwerk::OffensesFormatter)
       ).void
     end
-    def initialize(configuration: nil, out: $stdout, err_out: $stderr, style: OutputStyles::Plain.new)
+    def initialize(
+      configuration: nil,
+      out: $stdout,
+      err_out: $stderr,
+      style: OutputStyles::Plain.new,
+      offenses_formatter: nil
+    )
       @out = out
       @err_out = err_out
       @style = style
       @configuration = configuration || Configuration.from_path
       @progress_formatter = Formatters::ProgressFormatter.new(@out, style: style)
+      @offenses_formatter = offenses_formatter
```

Comment:
Let's move the default formatter to here and remove the need for the memoization in `offenses_formatter`